File: test/SBRequest_iPad/SBRequest_iPad/SearchViewController.m
//
//  SearchViewController.m
//  SBRequest_iPad
//
//  Created by caoyuan on 7/15/14.
//  Copyright (c) 2014 caoyuan. All rights reserved.
//

#import "SearchViewController.h"
#import "FuckHttp.h"
#import "AuthorizationViewController.h"
@interface SearchViewController () <UITextFieldDelegate>

@end

@implementation SearchViewController

- (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    NSLog(@"\n\n%@",self.token);
    self.searchTextField.delegate = self;

}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
 #pragma mark - Navigation

 // In a storyboard-based application, you will often want to do a little preparation before navigation
 - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
 {
 // Get the new view controller using [segue destinationViewController].
 // Pass the selected object to the new view controller.
 }
 */

- (IBAction)addNewWord:(id)sender {
    
     NSUserDefaults *userDefaultes = [NSUserDefaults standardUserDefaults];
    NSString *token =  [userDefaultes stringForKey:@"token"];
    NSLog(@"userDefaultes token : %@",token);
    
    NSString *addNewWordURL = [NSString stringWithFormat:@"https://api.shanbay.com/bdc/learning/?access_token=%@",token];

    NSDictionary *postDic = @{@"id":self.idd};
    NSMutableDictionary  *ddd = [[NSMutableDictionary alloc] initWithDictionary:postDic];
    NSDictionary *dic = (NSDictionary*)[FuckHttp post:addNewWordURL postDic:ddd];
    NSLog(@"\n\n%@",dic);
    NSString* msg = (NSString*)[dic objectForKey:@"msg"];
    if ([msg isEqualToString:@"SUCCESS"]) {
        self.statusBT.titleLabel.text = @"add success";
    }else{
        self.statusBT.titleLabel.text = @"failed";
    }
}

- (IBAction)search:(id)sender {
    NSLog(@"\n\n%@",self.token);
    if (self.searchTextField.text.length == 0) {
        [self.searchTextField becomeFirstResponder];
    }else {
        NSString *mayWithSpaceWord = self.searchTextField.text;
        self.word = [self getWordWithNoSpace:mayWithSpaceWord];
        NSDictionary *dic = (NSDictionary*)[FuckHttp get:[NSString stringWithFormat:@"https://api.shanbay.com/bdc/search/?word=%@",self.word]];
        NSLog(@"\n\n%@",dic);
        self.cn_definition = (NSString*)[[[dic objectForKey:@"data"] objectForKey:@"cn_definition"] objectForKey:@"defn"];
        self.resultLabel.text = self.cn_definition;
        self.idd = [NSString stringWithFormat:@"%@",[[dic objectForKey:@"data"] objectForKey:@"conent_id"] ];
    }
}

- (IBAction)statusBTAction:(id)sender {
    self.statusBT.titleLabel.text = @"status:";
    self.searchTextField.text = @"";
}

-(NSString*)getWordWithNoSpace:(NSString*) wordStr {
    NSString *word = wordStr;
    if (word) {
        while ([word hasSuffix:@" "]) {
            word = [word substringToIndex:([word length]-1)];
        }
    }
    return word;
}

#pragma mark - textfield delegate
-(BOOL)textFieldShouldReturn:(UITextField *)textField
{
    [self.searchTextField resignFirstResponder];
    [self search:[self getWordWithNoSpace:self.searchTextField.text]];
    return YES;
}
@end

File: test/ShanBay/ShanBay/SearchViewController.swift
//
//  SearchViewController.swift
//  ShanBay
//
//  Created by caoyuan on 7/11/14.
//  Copyright (c) 2014 caoyuan. All rights reserved.
//

import UIKit

class SearchViewController: UIViewController,UITextFieldDelegate {
    var token:NSString?
    var word:NSString?
    var idd:NSString?
    var cn_definition:NSString?
    @IBOutlet var statusBT : UIButton


    @IBOutlet var searchBt : UIButton
    @IBOutlet var serachTextField : UITextField
    @IBOutlet var resultLabel : UILabel

    init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
        // Custom initialization
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        println(token)

        // Do any additional setup after loading the view.
        self.serachTextField.delegate = self
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func search(sender : AnyObject) {
        println(token)
        if serachTextField.text.isEmpty {
            self.serachTextField.becomeFirstResponder()
//            self.searchBt.removeFromSuperview();
//            self.searchBt.frame = CGRectMake(26.0, 300.0, 46.0, 30.0)
//            self.view.addSubview(self.searchBt)
        }
        else{
            var mayWithSpaceWord = self.serachTextField.text
            word = getWordWithNoSpace(mayWithSpaceWord)!
            var dic = FuckHttp.get("https://api.shanbay.com/bdc/search/?word=\(word)")
            println(dic)
            cn_definition = dic?.objectForKey("data").objectForKey("cn_definition").objectForKey("defn") as? NSString
            self.resultLabel.text = cn_definition
            let idInt : AnyObject? = dic?.objectForKey("data").objectForKey("conent_id")
            self.idd = "\(idInt)"

        }
    }

//pragma mark - Navigation
// MARK: - UITableViewDataSource
    @IBAction func addNewWord(sender : AnyObject) {
        var addNewWordURL = "https://api.shanbay.com/bdc/learning/?access_token=\(token)"
        var postDic = ["id":self.idd!];
        var dic = FuckHttp.post(addNewWordURL, postData: postDic) as? NSDictionary
        println("\(dic)")
        let msg:NSString? = dic?.objectForKey("msg") as? NSString
        if msg!.isEqualToString("SUCCESS"){
            self.statusBT.titleLabel.text = "已经添加"
        }
        else{
             self.statusBT.titleLabel.text = "添加失败"
        }
    }

    @IBAction func statusBTAction(sender : AnyObject) {
        self.statusBT.titleLabel.text = "status:"
        self.serachTextField.text = ""
    }

    func textFieldShouldReturn(textField: UITextField!) {
        self.serachTextField.resignFirstResponder()
        search("nice")
    }

    // get the space out from the words
    func getWordWithNoSpace(WordStr:String?)->String?{
        var word = WordStr
        if word != nil {
            while word!.hasSuffix(" ") {
                word = word!.substringToIndex(countElements(word!)-1)
            }
        }
        return word!
    }


}
